policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py

```diff
@@ -0,0 +1,28 @@
+from policyengine_us.model_api import *
+
+
+class ct_social_security_benefit_adjustment(Variable):
+    value_type = float
+    entity = TaxUnit
+    unit = USD
+    label = "Connecticut social security benefit adjustment"
+    definition_period = YEAR
+    defined_for = StateCode.CT
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.ct.tax.income.subtractions.social_security
+        filing_status = tax_unit("filing_status", period)
+        taxable_ss = add(tax_unit, period, ["taxable_social_security"])
+
+        ss_rate = p.rate
+        total_ss = add(tax_unit, period, ["social_security"])
+        ss_fraction = total_ss * ss_rate
+        excess = tax_unit("ct_magi_excess_over_base", period)
```

Comment:
add comments on which lines those are from the worksheet and all relevant references that show the calculation